source-code-git/fs/src/interface.ts

```diff
@@ -8,17 +8,18 @@
  */
 export type NodeishFilesystem = {
 	writeFile: (path: string, data: string) => Promise<void>
-	readFile: (path: string, options: { encoding: "utf-8" }) => Promise<FileData>
+	/* NOTE: accept options as 'any' for now until we figure out how to get the node types to match */
+	readFile: (path: string, options?: any) => Promise<FileData>
```

Comment:
We can't set `readFile` options to any. Developers will call `fs.readFile(path)` and wonder why they don't receive a `string` as a return value. I saw that mistake in the wild already.

Reply:
Happened even to yourself :D

The node fs API is convoluted with a lot of overloads. I propose to use the latest [NodeJS 20 fs.promises API](https://nodejs.org/api/fs.html#fspromisesreadfilepath-options) with ONE WAY to do things. In most cases, this one way to do things follows an options as an object patten. 

- one way to do things increases DX
- reduces the amount of implementation work because we don't need conditional logic to check overloads

```ts
// the old way
fs.readFile("./", "utf-8")

// the new way
fs.readFile("./", { encoding: "utf-8" })

// the new wat offers additional flags without breaking changes [that we don't need to support now]
fs.readFile("./", { encoding: "utf-8", flag: "r" })
```

<img width="853" alt="CleanShot 2023-04-28 at 10 32 31@2x" src="https://user-images.githubusercontent.com/35429197/235097632-ebb3611c-d1b5-4f2c-a581-8a145efb7120.png">